Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -371,10 +408,11 @@ public Order GetOrderById(int orderId)
         private Order GetOrderByIdInternal(int orderId)
         {
             // this function can be invoked by brokerages when getting open orders, guard against null ref
-            if (_orders == null) return null;
+            if (_closedOrders == null || _openOrders == null) return null;
 
             Order order;
-            return _orders.TryGetValue(orderId, out order) ? order : null;
+            return _openOrders.TryGetValue(orderId, out order) ? order :
+                _closedOrders.TryGetValue(orderId, out order) ? order : null;
```

Comment:
hehe, prefer if we didn't use nested ternaries, although this is fairly legible for me, some other may disagree.

Reply:
done 👍